ogr/ogrsf_frmts/generic/ogrdatasource.cpp

```diff
@@ -105,6 +114,59 @@ int OGR_DS_GetSummaryRefCount(OGRDataSourceH hDataSource)
 /*                         OGR_DS_CreateLayer()                         */
 /************************************************************************/
 
+/**
+\brief This function attempts to create a new layer on the data source with the indicated name, coordinate system, geometry type.
+
+The papszOptions argument
+can be used to control driver specific creation options.  These options are
+normally documented in the format specific documentation.
+
+@deprecated Use GDALDatasetCreateLayer()
+
+ @param hDS The dataset handle.
+ @param pszName the name for the new layer.  This should ideally not
+match any existing layer on the datasource.
+ @param hSpatialRef handle to the coordinate system to use for the new layer,
+or NULL if no coordinate system is available. The driver might only increase
+the reference counter of the object to take ownership, and not make a full copy,
+so do not use OSRDestroySpatialReference(), but OSRRelease() instead when you
+are done with the object.
+ @param eType the geometry type for the layer.  Use wkbUnknown if there
+are no constraints on the types geometry to be written.
+ @param papszOptions a StringList of name=value options.  Options are driver
+specific, and driver information can be found at the following url:
+http://www.gdal.org/ogr_formats.html
```

Comment:
This URL is not correct. I am not sure if https://gdal.org/en/stable/drivers/vector/index.html is the right one.

Reply:
URL removed. This is deprecated API anyway

Reply:
The same URL appears in a couple of other files in this PR as well.